### PR TITLE
[stable9] Simplify testPermissionMovedGroupShare (#25573)

### PR DESCRIPTION
**Pull request by DeepDiver1975**
Reduces significantly the testing time but still cover the permission change

### CODE REVIEW

**Comment by PVince81** [comment]
@DeepDiver1975 many failures apparently ?

**Comment by DeepDiver1975** [comment]
@PVince81 this one is funny on php 5.4

````
PHP Fatal error:  Unsupported operand types in /ssd/jenkins/workspace/core-ci-linux-php5.4@3/database/sqlite/label/SLAVE/lib/private/share/share.php on line 1166
23:12:52 PHP Stack trace:
23:12:52 PHP   1. {main}() /usr/local/bin/phpunit:0
23:12:52 PHP   2. PHPUnit_TextUI_Command::main() /usr/local/bin/phpunit:537
23:12:52 PHP   3. PHPUnit_TextUI_Command->run() phar:///usr/local/bin/phpunit/phpunit/TextUI/Command.php:105
23:12:52 PHP   4. PHPUnit_TextUI_TestRunner->doRun() phar:///usr/local/bin/phpunit/phpunit/TextUI/Command.php:153
23:12:52 PHP   5. PHPUnit_Framework_TestSuite->run() phar:///usr/local/bin/phpunit/phpunit/TextUI/TestRunner.php:406
23:12:52 PHP   6. PHPUnit_Framework_TestSuite->run() phar:///usr/local/bin/phpunit/phpunit/Framework/TestSuite.php:722
23:12:52 PHP   7. PHPUnit_Framework_TestSuite->run() phar:///usr/local/bin/phpunit/phpunit/Framework/TestSuite.php:722
23:12:52 PHP   8. PHPUnit_Framework_TestCase->run() phar:///usr/local/bin/phpunit/phpunit/Framework/TestSuite.php:722
23:12:52 PHP   9. PHPUnit_Framework_TestResult->run() phar:///usr/local/bin/phpunit/phpunit/Framework/TestCase.php:699
23:12:52 PHP  10. PHP_Invoker->invoke() phar:///usr/local/bin/phpunit/phpunit/Framework/TestResult.php:607
23:12:52 PHP  11. call_user_func_array:{phar:///usr/local/bin/phpunit/php-invoker/Invoker.php:93}() phar:///usr/local/bin/phpunit/php-invoker/Invoker.php:93
23:12:52 PHP  12. PHPUnit_Framework_TestCase->runBare() phar:///usr/local/bin/phpunit/php-invoker/Invoker.php:93
23:12:52 PHP  13. PHPUnit_Framework_TestCase->runTest() phar:///usr/local/bin/phpunit/phpunit/Framework/TestCase.php:743
23:12:52 PHP  14. ReflectionMethod->invokeArgs() phar:///usr/local/bin/phpunit/phpunit/Framework/TestCase.php:866
23:12:52 PHP  15. Test_Files_Sharing_Mount->testPermissionMovedGroupShare() phar:///usr/local/bin/phpunit/phpunit/Framework/TestCase.php:866
23:12:52 PHP  16. OCP\Share::setPermissions() /ssd/jenkins/workspace/core-ci-linux-php5.4@3/database/sqlite/label/SLAVE/apps/files_sharing/tests/sharedmount.php:308
23:12:52 PHP  17. OC\Share\Share::setPermissions() /ssd/jenkins/workspace/core-ci-linux-php5.4@3/database/sqlite/label/SLAVE/lib/public/share.php:332

**Comment by PVince81** [comment]
no idea...

**Comment by DeepDiver1975** [comment]
````
There were 2 errors:

1) Test_Files_Sharing_Mount::testPermissionMovedGroupShare with data set #0 ('file', true, true)
Error: Unsupported operand types

/ssd/jenkins/workspace/ownCloud/core/PR-25608/lib/private/share/share.php:1166
/ssd/jenkins/workspace/ownCloud/core/PR-25608/lib/public/share.php:332
/ssd/jenkins/workspace/ownCloud/core/PR-25608/apps/files_sharing/tests/sharedmount.php:308

2) Test_Files_Sharing_Mount::testPermissionMovedGroupShare with data set #1 ('folder', true, true)
Error: Unsupported operand types

/ssd/jenkins/workspace/ownCloud/core/PR-25608/lib/private/share/share.php:1166
/ssd/jenkins/workspace/ownCloud/core/PR-25608/lib/public/share.php:332
/ssd/jenkins/workspace/ownCloud/core/PR-25608/apps/files_sharing/tests/sharedmount.php:308

File: apps/files_sharing/tests/sharedmount.php [hunk 1]
```diff
@@ -240,63 +240,27 @@ function dataProviderTestStripUserFilesPath() {
...
-		$permissions = [
-			\OCP\Constants::PERMISSION_UPDATE,
-			\OCP\Constants::PERMISSION_CREATE,
-			\OCP\Constants::PERMISSION_SHARE,
-			\OCP\Constants::PERMISSION_DELETE,
+		return [
+			[
+				'file',
+				\OCP\Constants::PERMISSION_READ
+				|| \OCP\Constants::PERMISSION_UPDATE
```
Comment (PVince81) [inline apps/files_sharing/tests/sharedmount.php]: ah yes of course, it should be `|`... I did it wrong on master, let's close and I'll resubmit...

**Comment by PVince81** [comment]
I tested it wrong, see comment above.

Will fix on master and stable9.1 first.

**Comment by PVince81** [comment]
Grrr, are you kidding me...

```
Test_Files_Sharing_Mount::testPermissionMovedGroupShare with data set #0 ('file', true, true)
Error: Unsupported operand types

/var/lib/jenkins/workspace/owncloud-core/core/PR-25608/lib/private/share/share.php:1166
/var/lib/jenkins/workspace/owncloud-core/core/PR-25608/lib/public/share.php:332
/var/lib/jenkins/workspace/owncloud-core/core/PR-25608/apps/files_sharing/tests/sharedmount.php:308
```

**Comment by PVince81** [comment]
Aha, I forgot to fix the operands on this branch. Fixed now and rebased.

Jenkins, pass now!

**Pull request merged**